### PR TITLE
Use grails-bom, ContainerGebSpec, cleanup dependencies and fix tests

### LINKED ISSUE #755: gorm-mongodb:9.0.0-SNAPSHOT Failing Tests
Requires 

4 tests fail with error similar to 

```
Test GORM access
java.lang.IllegalStateException: Either class [functional.tests.Book] is not a domain class or GORM has not been initialized correctly or has already been shutdown. Ensure GORM is loaded and configured correctly before calling any methods on a GORM entity.
	at org.grails.datastore.gorm.GormEnhancer.stateException(GormEnhancer.groovy:467)
	at org.grails.datastore.gorm.GormEnhancer.findInstanceApi(GormEnhancer.groovy:315)
	at org.grails.datastore.gorm.GormEnhancer.findInstanceApi(GormEnhancer.groovy:312)
	at org.grails.datastore.gorm.GormEntity$Trait$Helper.currentGormInstanceApi(GormEntity.groovy:1366)
	at org.grails.datastore.gorm.GormEntity$Trait$Helper.save(GormEntity.groovy:153)
	at functional.tests.BookUnitSpec.Test GORM access(BookUnitSpec.groovy:30)
```

- Currently waiting on Grails 7 version of Spring Security Rest, already updated in Spring Security Core, but Rest is pulling old version
- https://github.com/search?q=org%3Agrails%20Pbkdf2PasswordEncoder&type=code

Could not find which method <init>() to invoke from this list: public org.springframework.security.crypto.password.Pbkdf2PasswordEncoder#<init>(java.lang.CharSequence, int, int, int) public org.springframework.security.crypto.password.Pbkdf2PasswordEncoder#<init>(java.lang.CharSequence, int, int, org.springframework.security.crypto.password.Pbkdf2PasswordEncoder$SecretKeyFactoryAlgorithm) at app//grails.plugin.springsecurity.SpringSecurityCoreGrailsPlugin.idToPasswordEncoder(SpringSecurityCoreGrailsPlugin.groovy:1125) at grails.plugin.springsecurity.SpringSecurityCoreGrailsPlugin.doWithSpring_closure12(SpringSecurityCoreGrailsPlugin.groovy:492) at app//groovy.lang.Closure.call(Closure.java:433) at app//groovy.lang.Closure.call(Closure.java:412) at app//grails.spring.BeanBuilder.invokeBeanDefiningClosure(BeanBuilder.java:759) at app//grails.spring.BeanBuilder.beans(BeanBuilder.java:588) at app//grails.spring.BeanBuilder.invokeMethod(BeanBuilder.java:531) at

remove -x test -x integrationTest from gradle.yml and groovy-joint-workflow.yml

### PR DESCRIPTION
**Pull request by jamesfredley**
This will get us ready for M1.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -93,37 +83,21 @@ subprojects { Project subproject ->
 
         boolean usesGeb = project.name.contains('grails3-hibernate5-mongodb') || project.name.contains('grails3-mongodb') || project.name.contains('grails3-mongodb-database-per-tenant')|| project.name.contains('grails3-rxmongodb')
```
Comment (jdaugherty) [inline build.gradle]: Is this still used?

**Review by jdaugherty — APPROVED** [review]

**Comment by matrei** [comment]
@jamesfredley We currently have failing test. Do we want to address those before merging, or is there another plan?

**Comment by jamesfredley** [comment]
@matrei They popped up this morning when I moved a few things over to grails-bom -> spring-boot-dependencies.  I will track it down and resolve before we merge.

File: build.gradle (hunk 1 above)
Comment (jamesfredley) [reply build.gradle]: It was not and I have removed it.

File: build.gradle [hunk 2]
```diff
@@ -91,39 +81,21 @@ subprojects { Project subproject ->
...
-                } else if (details.requested.group == "org.springframework.boot") {
-                    details.useVersion(springBootVersion)
-                }
-            }
-        }
-
         dependencies {
-            testImplementation "jakarta.annotation:jakarta.annotation-api:$jakartaAnnotationApiVersion"
+            implementation platform("org.grails:grails-bom:$grailsVersion")
+            implementation "jakarta.servlet:jakarta.servlet-api"
```
Comment (matrei) [inline build.gradle]: compileOnly?

**Review by matrei — APPROVED** [review]

File: build.gradle (hunk 2 above)
Comment (jamesfredley) [reply build.gradle]: Good catch, I was able to remove it entirely, now that the test applications have updated dependencies.

Comment (matrei) [reply build.gradle]: @jamesfredley I did not go through all the dependencies, to much of a maze for a quick review. I don't like when dependencies are assigned to subprojects with conditions. It becomes really hard to untangle. But we save that for another day 🌞

**Pull request merged**